future/transaction.go

```diff
@@ -62,20 +80,7 @@ func MakePaymentTxn(from, to string, amount uint64, note []byte, closeRemainderT
 		},
 	}
 
-	// Update fee
-	if !params.FlatFee {
-		eSize, err := transaction.EstimateSize(tx)
-		if err != nil {
-			return types.Transaction{}, err
-		}
-		tx.Fee = types.MicroAlgos(eSize * uint64(params.Fee))
-	}
-
-	if tx.Fee < MinTxnFee {
-		tx.Fee = MinTxnFee
-	}
-
-	return tx, nil
+	return setFee(tx, params)
```

Comment:
In this case, if the flatFee is set to a higher value than the MinTxnFee, the fee will be higher than before this change. Right?

Reply:
You are correct about the intention, if `FlatFee = true` then we should use `max(param.Fee, MinTxnFee)`

This was actually the behavior before as well because the fee is always initialized to params.Fee. In `setFee` I'm setting it a second time just to be sure.